lib/scope.js

```diff
@@ -712,7 +712,9 @@ module.exports = {
           sought_var: page_data.sought_var, pattern_data: pattern,
         });
         // If that works, it's the right map for the proxy values
-        if ( maybe_map ) { proxies_to_names_map = maybe_map; }
+        if ( Object.keys( maybe_map ).length ) {
+          proxies_to_names_map = maybe_map;
```

Comment:
Do we expect to have valid `maybe_map` values multiple times in this loop? If not, we can break here, since this code makes sure only one value of `maybe_map` gets written to `proxies_to_names_map`.

Reply:
The loop is needed for setting up the nodes, so we can't break. Regarding the possibility of multiple values, I've made #206.